mbed_greentea/mbed_test_api.py

```diff
@@ -174,12 +174,28 @@ def run(self):
 
         def stop(self):
             self.active = False
+
+            # Try stopping mbed-host-test
             try:
-                self.proc.terminate()
-            except Exception as e:
-                print "ProcessObserver.stop(): %s" % str(e)
+                self.proc.stdin.close()
+            finally:
                 pass
 
+            # Give 5 sec for mbedhtrun to exit
+            for i in range(5):
```

Comment:
This would be bit confusing. But I realise that ret_code should be declared outside for loop.